cantools/database/utils.py

```diff
@@ -205,3 +209,26 @@ def create_little():
     return Formats(big_compiled,
                    little_compiled,
                    big_padding_mask & little_padding_mask)
+
+
+def sawtooth_to_network_bitnum(sawtooth_bitnum):
+    '''Convert SawTooth bit number to Network bit number
+
+    Byte     |   0   |   1   |
+    Sawtooth |7 ... 0|15... 8|
+    Network  |0 ... 7|8 ...15|
+    '''
+    return (8 * (sawtooth_bitnum // 8)) + (7 - (sawtooth_bitnum % 8))
+
+
+def cdd_offset_to_dbc_start_bit(cdd_offset, bit_length, byte_order):
+    '''Convert CDD/c-style field bit offset to DBC field start bit convention.
+
+    BigEndian (BE) fields are located by their MSBit's sawtooth index.
+    LitteleEndian (LE) fields located by their LSBit's sawtooth index.
+    '''
+    if byte_order == "big_endian":
+        # Note: Allow for BE fields that are smaller or larger than 8 bits.
+        return (8 * (cdd_offset // 8)) + min(7, (cdd_offset % 8) + bit_length - 1)
```

Comment:
I use Vector dbc file and find the start bit is wrong with Motorola Big End.

Start bit in motorola signals from Vector dbc files

https://github.com/reinzor/libcan-encode-decode/issues/1

https://github.com/ebroecker/canmatrix/wiki/signal-Byteorder
```python
def itom_sb(b, length):
    b = b - (b % 8) + 7 - (b % 8)
    b = b + length - 1
    b = b - (b % 8) + 7 - (b % 8)
    return b

>>> itom_sb(7, 32) == 24
```

Reply:
Hi there. You are posting to a PR that refers to CCD file format. Do you use CCD?